tv_apps/smart-system/js/accessibility.js

```diff
@@ -558,30 +551,27 @@
         this.cancel();
       }
 
-      var sentence = this.buildUtterance(aData);
-      if (!sentence) {
-        if (aCallback) {
-          aCallback();
-        }
-        return;
-      }
-
-      var utterance = new this.utterance(sentence);
-      utterance.volume = aVolume;
-      utterance.rate = aRate;
-      utterance.addEventListener('end', function() {
-        if (this.captions) {
-          this.hideSpeech();
-        }
-        if (aCallback) {
-          aCallback();
+      return this.buildUtterance(aData).then(sentence => {
+        if (!sentence) {
+          return Promise.resolve();
```

Comment:
Uhh, I'm not sure. System's accessibility.js doesn't have `aCallback`. I'm not sure if it's a left over and has been refactored in System, or an addition in TV System.

What's your preference?